src/som/compiler/KernanClient.java

```diff
@@ -178,9 +166,12 @@ private void setMessage(final String message) {
    * @return - an instance of {@link Frame}
    */
   public Frame buildFrame(final int operationCode, final String message) {
-    Frame frame = new Frame(message.length());
+    byte[] bytes = null;
+    try { message.getBytes("UTF8"); }
+    catch (java.io.UnsupportedEncodingException e) { assert false; }
```

Comment:
One more thing: this here is problematic. Swallowing exceptions like this is going to make for very unhappy users. For some reason, nothing happens if things go wrong. That's hard to diagnose.

I'd propose to wrap the exception into a RuntimeException instead. Then at least users will get a clue what's going wrong.